src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -264,6 +264,15 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
                 Dispatcher.UIThread.Post(this.BringIntoView); // must use the Dispatcher, otherwise the TreeView doesn't scroll
             }
         }
+        
+        protected virtual void HeaderDoubleTapped(object? sender, TappedEventArgs e)
```

Comment:
Keep in mind when you are adding a method like this you are extending the API that may be used in derived controls (which is what you need it for). Therefore, we don't want to simply make the event handler public -- we want to make the event handler call a protected method that becomes part of the control's API.

Convention is usually to do the following:

 1. The event handler signature stays the same but we now call into a protected virtual method that can be overridden by derived classes.
 2. The protected virtual method uses the 'On[Event]' naming convention
 3. The protected virtual method doesn't need the sender parameter -- it is known since we are within an instance of TreeViewItem itself.
 4. All of this said, this convention is usually for invoking an event. So code would call OnHeaderDoublePressed() which would then invoke an event. So the logic is kind-of reversed here. Still seems the way to go though unless others have different ideas.

It's also good to add comments to any API.

```csharp
        private void HeaderDoubleTapped(object? sender, TappedEventArgs e)
        {
            OnHeaderDoubleTapped(e);
        }

        /// <summary>
        /// Invoked when the <see cref="DoubleTapped"/> event occurs in the header.
        /// </summary>
        protected virtual void OnHeaderDoubleTapped(TappedEventArgs e)
        {
            if (ItemCount > 0)
            {
                IsExpanded = !IsExpanded;
                e.Handled = true;
            }
        }
```